token-refresh.sh

```diff
@@ -6,14 +6,43 @@
 # Note: you must be logged into the cloud foundry CLI
 # for this script to work
 
-file=".env.local"
 
-cf_res=$(cf oauth-token)
-token=${cf_res#* }
+refresh_token() {
+    file=".env.local"
+    cf_res=$(cf oauth-token)
+    token=${cf_res#* }
+    if [ "$token" == "FAILED" ]; then
+        echo "Failed to obtain token"
+        return 1
+    else
+        sed -i '' -e "s/CF_API_TOKEN=.*/CF_API_TOKEN=${token}/" $file
+        echo "$(date): Updated token in ${file}"
+    fi
+}
 
-if [ "$token" == "FAILED" ]; then
-  exit 1
-else
-  sed -i '' -e "s/CF_API_TOKEN=.*/CF_API_TOKEN=${token}/" $file
-  echo "Added token to ${file}"
-fi
+# Function to cleanup background process
+cleanup() {
+    echo "Cleaning up token refresh process..."
+    if [ -f .token-refresh.pid ]; then
+        kill $(cat .token-refresh.pid)
+        rm .token-refresh.pid
+    fi
+    exit 0
+}
+
+# Set up trap to catch Ctrl+C and other termination signals
+trap cleanup SIGINT SIGTERM
+
+# Initial token refresh
+refresh_token
+
+# Start token refresh loop in background
+(
+    while true; do
+        sleep 900  # 15 minutes in seconds
```

Comment:
Should this be slightly less than 15 minutes to catch it before it expires?

Reply:
I was wondering about that, but didn't know if that was being pedantic. how's 850?